supervisor.go

```diff
@@ -0,0 +1,193 @@
+	supervisorSuspendAllPathPrefix   = "druid/indexer/v1/supervisor/suspendAll"
+	supervisorResumePathPrefix       = "druid/indexer/v1/supervisor/:supervisorId/resume"
+	supervisorResumeAllPathPrefix    = "druid/indexer/v1/supervisor/resumeAll"
+	supervisorResetPathPrefix        = "druid/indexer/v1/supervisor/:supervisorId/reset"
+	supervisorTerminatePathPrefix    = "druid/indexer/v1/supervisor/:supervisorId/terminate"
+	supervisorTerminateAllPathPrefix = "druid/indexer/v1/supervisor/terminateAll"
+	supervisorShutdownPathPrefix     = "druid/indexer/v1/supervisor/:supervisorId/shutdown"
+)
+
+// SupervisorService is a service that submits ingestion tasks to druid supervisor API.
+type SupervisorService struct {
+	client *Client
+}
+
+// CreateSupervisorResponse is a response object of Druid SupervisorService's SubmitTask method
+type CreateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+// TerminateSupervisorResponse is a response object of Druid SupervisorService's Terminate method
+type TerminateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+type GetStatusResponsePayload struct {
+	Datasource      string `json:"dataSource"`
+	Stream          string `json:"stream"`
+	State           string `json:"state"`
+	Partitions      int    `json:"partitions"`
+	Replicas        int    `json:"replicas"`
+	DurationSeconds int    `json:"durationSeconds"`
+}
+
+// GetStatusResponse is a response object of Druid SupervisorService's GetStatus method
+type GetStatusResponse struct {
+	SupervisorId   string                   `json:"id"`
+	GenerationTime time.Time                `json:"generationTime"`
+	Payload        GetStatusResponsePayload `json:"payload"`
+}
+
+// CreateOrUpdate submits an ingestion specification to druid Supervisor API with a pre-configured druid client
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#create-or-update-a-supervisor
+func (s *SupervisorService) CreateOrUpdate(spec interface{}) (string, error) {
+	r, err := s.client.NewRequest("POST", supervisorPathPrefix, spec)
+	if err != nil {
+		return "", err
+	}
+	var result CreateSupervisorResponse
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return "", err
+	}
+	return result.SupervisorId, nil
+}
+
+func applySupervisorId(input string, supervisorId string) string {
+	return strings.Replace(input, ":supervisorId", supervisorId, 1)
+}
+
+// GetActiveIDs returns array of active supervisor IDs
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-ids
+func (s *SupervisorService) GetActiveIDs() ([]string, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method GetActiveIDs not implemented")
+}
+
+// GetAllActiveSpecs returns array of active supervisor objects
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-objects
+func (s *SupervisorService) GetAllActiveSpecs() ([]interface{}, error) {
```

Comment:
```suggestion
func (s *SupervisorService) GetAllActiveSpecs() ([]any, error) {
```

use `any` here elsewhere as it's preferred way with go 1.18 +

Reply:
Replaced with explicit types here and elsewhere.